src/plugins/typechecker/README.md

```diff
@@ -0,0 +1,115 @@
+- infos = Information about the typechecker plugin is in keys below
+- infos/author = e1528532 <e1528532@libelektra.org>
+- infos/licence = BSD
+- infos/needs = 
+- infos/provides = typechecker
+- infos/recommends =
+- infos/placements = postgetstorage presetstorage
+- infos/status = maintained experimental global
+- infos/metadata =
+- infos/description = a plugin which typechecks configuration specifications
+
+## Introduction
+
+A plugin which typechecks specifications before setting keys and after getting keys from
+a mounted specification.
+
+The typesystem is currently based on regular expressions. Each key is assigned with a regex
+that describes its contents. Links between keys can only be done if the regexes describing 
+the linked keys are compatible with each other.
```

Comment:
What means "compatible". Is this only relevant for links?

Reply:
elaborated a bit more